neptune-core/src/application/util_proof/sent/mod.rs

```diff
@@ -0,0 +1,329 @@
+    utxo_digest: Digest,
+}
+
+pub fn claim_inputs(
+    c: Claim,
+    receiver_digest: Digest,
+    release_date: crate::api::export::Timestamp,
+) -> Claim {
+    c.with_input(
+        [
+            [release_date.0].as_slice(),
+            receiver_digest.reversed().values().as_slice(),
+        ]
+        .concat(),
+    )
+}
+pub fn claim_outputs(
+    c: Claim,
+    sender_randomness_digest: Digest,
+    aocl_digest: Digest,
+    lock_postimage: Digest,
+    amount: NativeCurrencyAmount,
+    // amount_timelocked: NativeCurrencyAmount,
+) -> Claim {
+    c.with_output(
+        [
+            lock_postimage.values().into(),
+            sender_randomness_digest.values().to_vec(),
+            aocl_digest.values().into(),
+            amount.encode(),
+            // amount_timelocked.encode(),
+        ]
+        .concat(),
+    )
+}
+
+fn library_and_code() -> (Library, Vec<LabelledInstruction>) {
+    let u64_stack_size: u32 = tasm_lib::prelude::DataType::U64
+        .stack_size()
+        .try_into()
+        .unwrap();
+
+    let mut library = Library::new();
+    let lib_hash_varlen = library.import(Box::new(
+        tasm_lib::hashing::algebraic_hasher::hash_varlen::HashVarlen,
+    ));
+    let lib_ms_commit = library.import(Box::new(tasm_lib::neptune::mutator_set::commit::Commit));
+    let lib_mmr_verify = library.import(Box::new(
+        tasm_lib::mmr::verify_from_memory::MmrVerifyFromMemory,
+    ));
+    let lib_bagpeaks = library.import(Box::new(tasm_lib::mmr::bag_peaks::BagPeaks));
+    let release_date = library.kmalloc(1);
+    let lib_add_all_amounts_and_check_time_lock = library.import(
+        Box::new(crate::protocol::consensus::type_scripts::amount::add_all_amounts_and_check_time_lock::AddAllAmountsAndCheckTimeLock {
+            digest_source: crate::protocol::consensus::type_scripts::amount::total_amount_main_loop::DigestSource::
+            Hardcode(crate::protocol::consensus::type_scripts::native_currency::NativeCurrency.hash()),
+            release_date,
+    }));
+
+    // let rustfield_membershipproof = rustfield!(WitnessMemory::membership_proof);
+    let rustfield_senderrandomness = rustfield!(MsMembershipProof::sender_randomness);
+    let rustfield_utxo = rustfield!(WitnessMemory::utxo);
+    let rustfield_utxodigest = rustfield!(WitnessMemory::utxo_digest);
+    let rustfield_aoclleafindex = rustfield!(MsMembershipProof::aocl_leaf_index);
+
+    let payload = triton_asm! {
+        /* pasted from <../reserves>
+        ============== */
+        // _
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        {&rustfield_utxo}
+        {&Utxo::get_field("lock_script_hash")}
+        read_mem 5
+        pop 1
+        // [lock_script_digest]
+        write_io 5
+        // _
+
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        // *w
+        {&WitnessMemory::get_field("aocl")}
+        // *aocl
+
+        // prepare the value for '7.'
```

Comment:
Please make the files self-contained. Don't make the reader open another file to figure out what "7." means.